select.go

```diff
@@ -13,85 +13,155 @@ import (
-	Next     Key // Next defaults to down arrow key
-	Prev     Key // Prev defaults to up arrow key
-	PageUp   Key // PageUp defaults to left arrow key
-	PageDown Key // PageDown defaults to right arrow key
-	Search   Key // Search defaults to '/' key
+	// Next is the key used to move to the next element inside the list. Defaults to down arrow key.
+	Next     Key
+
+	// Prev is the key used to move to the previous element inside the list. Defaults to up arrow key.
+	Prev     Key
+
+	// PageUp is the key used to jump back to the first element inside the list. Defaults to left arrow key.
+	PageUp   Key
+
+	// PageUp is the key used to jump forward to the last element inside the list. Defaults to right arrow key.
+	PageDown Key
+
+	// Search is the key used to trigger the search mode for the list. Default to the "/" key.
+	Search   Key
 }
 
-// Key defines a keyboard code and a display representation for the help
-// Check https://github.com/chzyer/readline for a list of codes
+// Key defines a keyboard code and a display representation for the help menu.
 type Key struct {
+	// Code is a rune that will be used to compare against typed keys with readline.
+	// Check https://github.com/chzyer/readline for a list of codes
 	Code    rune
+
+	// Display is the string that will be displayed inside the help menu to help inform the user
+	// of which key to use on his keyboard for various functions.
 	Display string
 }
 
-// SelectTemplates allow a select prompt to be customized following stdlib
-// text/template syntax. If any field is blank a default template is used.
+// SelectTemplates allow a select list to be customized following stdlib
+// text/template syntax. Custom state, colors and background color are available for use inside
+// the templates and are documented inside the Variable section of the docs.
 type SelectTemplates struct {
-	// Active is a text/template for the label.
+	// Label is a text/template for the main command line label. Defaults to printing the label as it with
+	// the IconInitial.
 	Label string
 
-	// Active is a text/template for when an item is current active.
+	// Active is a text/template for when an item is currently active within the list.
 	Active string
 
-	// Inactive is a text/template for when an item is not current active.
+	// Inactive is a text/template for when an item is not currently active inside the list. This
+	// template is used for all items unless they are active or selected.
 	Inactive string
 
 	// Selected is a text/template for when an item was successfully selected.
 	Selected string
 
 	// Details is a text/template for when an item current active to show
 	// additional information. It can have multiple lines.
+	//
+	// Detail will always be displayed for the active element and thus can be used to display additional
+	// information on the element beyond its label.
+	//
+	// To make this field multi-line, a literal template must be used (``). Prompt-ui will not trim spaces
```

Comment:
you can insert newline chars in any string.

Reply:
Good to know, I got my javascript mixed in